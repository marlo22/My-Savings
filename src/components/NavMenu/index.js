import React from 'react';
import styled from 'styled-components';

import { GlobalContextConsumer } from '../../context/Global';

import { Container } from 'native-base';
import { NavMenuItem } from '../';

import logout from '../../api/logout';

import { secondaryColor } from '../../themes';

const NavMenuWrapper = styled(Container)`
  background-color: ${secondaryColor};
  color: #ffffff;
  width: 80%;
  position: absolute;
  top: 0;
  left: 0;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.2);
`;

const NavMenu = () => {
  const menuItems = [
    { text: 'Pulpit', iconName: 'home', onPress: () => console.warn('go to dashboard') },
    { text: 'Limity', iconName: 'timer', onPress: () => console.warn('go to limits') },
    { text: 'Wydatki', iconName: 'cash', onPress: () => console.warn('go to spendings') },
    { text: 'Kategorie', iconName: 'folder-open', onPress: () => console.warn('go to categories') },
    { text: 'Statystyki', iconName: 'stats', onPress: () => console.warn('go to stats') },
    { text: 'Ustawienia', iconName: 'settings', onPress: () => console.warn('go to settings') },
    { text: 'Wyloguj', iconName: 'person', onPress: logout }
  ];

  return (
    <GlobalContextConsumer>
      {context => context.isNavOpen ? (
        <NavMenuWrapper>
          {menuItems.map(item => <NavMenuItem key={item.text} {...item} />)}
        </NavMenuWrapper>
      ) : null}
    </GlobalContextConsumer>
  );
}

export default NavMenu;