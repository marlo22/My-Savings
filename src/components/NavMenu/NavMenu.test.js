import React from 'react';
import { shallow } from 'enzyme';

import NavMenu from './';

jest.mock('../../api/logout', () => jest.fn());

const toggleNav = jest.fn();

jest
  .spyOn(React, 'useContext')
  .mockImplementationOnce(() => ({
    isNavOpen: true,
    toggleNav
  }));

const setup = () => shallow(
  <NavMenu />
);

let component = setup();

describe('<NavMenu />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('click menu item should call toggleNav context\'s function', () => {
    const menuItem = component.find('NavMenuItem').last();
    menuItem.simulate('press');
    setTimeout(() => expect(toggleNav).toHaveBeenCalled());
  });

  test('if nav is close should return null', () => {
    jest
      .spyOn(React, 'useContext')
      .mockImplementationOnce(() => ({
        isNavOpen: false,
        toggleNav
      }));

    component = setup();

    expect(component.html()).toBe(null);
  });

  // @TODO - test asynchronous onPressProxy
});