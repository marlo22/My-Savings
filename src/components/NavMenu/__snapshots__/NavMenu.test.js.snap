// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenu /> snapshot 1`] = `
<Styled(Styled(Container))
  elevation={99}
>
  <NavMenuItem
    iconName="home"
    key="Pulpit"
    onPress={[Function]}
    text="Pulpit"
  />
  <NavMenuItem
    iconName="timer"
    key="Limity"
    onPress={[Function]}
    text="Limity"
  />
  <NavMenuItem
    iconName="cash"
    key="Wydatki"
    onPress={[Function]}
    text="Wydatki"
  />
  <NavMenuItem
    iconName="folder-open"
    key="Kategorie"
    onPress={[Function]}
    text="Kategorie"
  />
  <NavMenuItem
    iconName="stats"
    key="Statystyki"
    onPress={[Function]}
    text="Statystyki"
  />
  <NavMenuItem
    iconName="settings"
    key="Ustawienia"
    onPress={[Function]}
    text="Ustawienia"
  />
  <NavMenuItem
    iconName="person"
    key="Wyloguj"
    onPress={[Function]}
    text="Wyloguj"
  />
</Styled(Styled(Container))>
`;
